docs/philosophy/outcomes/developer_adoption.md

```diff
@@ -269,72 +210,72 @@ The simplest way to calculate _operational efficiency_ is by comparing the numbe
 
 #### Data Points
 
-_Adoption events ($E$)_
-
-:   A unique target user does something valuable using the tool
-
-    Expressed as a set of timestamps
+[_Adoption events $(E)$_](#adoption-events)
 
 _Component Maintainers ($M$)_
 
-:   A count of the number of people it takes to maintain the component
+:   The number of people it takes to maintain the component
 
     Expressed as a number
 
+    TODO this is constant or can increase/decrease over time?
+
 #### Formulas
 
+[_Number of adoption events at a given time interval $(E(t,\Delta t))$_](#number-of-adoption-events-at-a-given-time-interval)
+
+_Number of maintainers at a given time interval $(M(t,\Delta t))$_
+
+:   Given a timestamp $t$ and a time interval $t-\Delta t$, the number maintainers between $t$ and $t-\Delta t$, is calculated as follows:
+
+    $$
+    M(t,\Delta t) = \text{count}(\{(\text{maintainer},x) : \exists\ x,t -\Delta t \leq x \leq t, (\text{maintainer},x) \in M\})
+    $$
+
 !!! formula ""
 
-    _Operational Efficiency ($OE$)_
+    _Operational Efficiency $(OE(t,\Delta t))$_
 
-    :   Given the total number of adoption events $E$ and the number of maintainers $M$ supporting the component, we calculate operational efficiency $OE$ using a simple ratio:
+    :   The operational efficiency at timestamp $t$ over time interval $\Delta t$
 
         $$
-        OE = \frac{E_{T_0}}{M_{T_0}}
+        OE(t) = \frac{E(t,\Delta t)}{M(t,\Delta t)}
         $$
 
 ### :material-account-heart-outline: __Developer Satisfaction__
 
-Measures the extent to which the platform is meeting the needs and wants of developers
+Measures the extent to which the platform is meeting the needs and wants of developers.
 
 #### Data Points
 
-_Net Promoter Score survey results ($R$)_
+[_Number of active users at a given time interval $(U(t,\Delta t))$_](#number-of-active-users-at-a-given-time-interval)
+
+_Net Promoter Score survey $(S)$_
 
 :   The set of scores returned from a net promoter score survey
 
     Expressed as a set of integers between 0 and 10
 
-_Number of current active users ($U_{T_0}$)_
-
-:   The number of users who have at least one adoption event between timestamp $T_0$ and $T_1$.
-
-    Expressed as an integer (calculated in previous measures)
-
 #### Formulas
 
-_Number of survey results_
+_Survey response rate $(SR(t,\Delta t))$_
 
-:   The number of people that responded to the net promoter score survey
+:   The percentage of the active users between $t$ and $t-\Delta t$ whose experiences were captured by the survey, is calculated as follows:
 
     $$
-    P = count(R)
+    SR(t,\Delta t) = \frac{\text{count}(S)}{U(t,\Delta t)} \cdot 100
     $$
 
-_Response rate_
-
-:   The percentage of the active users whose experiences were captured by the survey
-
-    $$
-    RR = \frac{ P }{ U_{T_0} } \cdot 100
-    $$
+TODO Net Promoter Score survey (S) results is constant or can increase/decrease over time?
```

Comment:
Similar question about maintainers: if we track the changes Net Promoter Score survey (S) had over time or just care about its value at the current time

Reply:
similar, it can change as more people fill out the survey and it won't really effect the ability to measure. as long as there is at least one response you can calculate NPS. we just want to continuously know how many results we have at a time when we calculate